daemon/emer-main.c

```diff
@@ -134,6 +151,58 @@ on_upload_events (EmerEventRecorderServer *server,
   return TRUE;
 }
 
+typedef enum {
+  AUTHORIZED_METHOD_NONE = 0,
+  AUTHORIZED_METHOD_SET_ENABLED,
+  AUTHORIZED_METHOD_RESET_TRACKING_ID
+} AuthorizedMethod;
+
+static const gchar *not_authorized_error_messages[] = {
+  NULL,
+  "Disabling metrics is only allowed from system settings",
+  "Only privileged users can reset the tracking id",
+};
+static const gsize n_not_authorized_error_messages = G_N_ELEMENTS (not_authorized_error_messages);
+
+static const gchar *
+not_authorized_error_message_for_method (AuthorizedMethod method)
```

Comment:
This method is never used!

I think a better approach here would be to define:

```c
typedef struct {
    const gchar *method_name;
    const gchar *method_full_name;
    const gchar *error_message;
} AuthorizedMethod;

static const AuthorizedMethods[] authorized_methods = {
   { "SetEnabled", "com.endlessm.Metrics.SetEnabled", "..." },
   { "ResetTrackingId", ... },
   { NULL }
};

static const AuthorizedMethod *
lookup_authorized_method (const gchar *method_name)
{
  // obvious loop, return NULL if not found
}
```

Then `on_authorize_method_check` would call this function once, early-return if it returned NULL, and then have all the other information it needed right there. No need for extra lookups and `g_assert`s.
```

Reply:
Makes sense, I think that's a lot simpler.